packages/db/src/models/granule.ts

```diff
@@ -5,30 +5,79 @@ import { tableNames } from '../tables';
 
 import { PostgresGranule, PostgresGranuleRecord } from '../types/granule';
 
+import { GranuleExecutionHistoryPgModel } from './granule-execution-history';
+
 export default class GranulePgModel extends BasePgModel<PostgresGranule, PostgresGranuleRecord> {
   constructor() {
     super({
       tableName: tableNames.granules,
     });
   }
 
-  upsert(
+  async createWithExecutionHistory(
```

Comment:
I don't know if there really was/is a legit scenario where we want to write a granule without an execution. I don't think there is one, because the whole value of a granule-execution history is non-existent if you're not actually ... writing that history. 

Yeah, I wanted to have this extend `create` from the base model, but you can't do that if the function signature is not the same. And since this method needs to accept an `executionCumulusId` now, that can't work. 

See https://stackoverflow.com/questions/50729485/override-method-with-different-argument-types-in-extended-class-typescript

Reply:
TODO: write granule lib/helper that handles granule table write and granule/executions writes. re-use this in data-migration and API lambda writes

Reply:
6ba78f18dcea6ffb348a4b407e337bd4fbc5d068
7b3eb6f3e607be4aa8697c99be15b0e1ae116254